pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -185,7 +187,13 @@ func newAlertmanager(ctx context.Context, orgID int64, cfg *setting.Cfg, store s
 
 	am.wg.Add(1)
 	go func() {
-		am.silences.Maintenance(15*time.Minute, silencesFilePath, am.stopc, func() (int64, error) {
+		am.silences.Maintenance(silenceMaintenanceInterval, silencesFilePath, am.stopc, func() (int64, error) {
+			// Delete silences older than the retention period.
+			if _, err := am.silences.GC(); err != nil {
+				return 0, err
```

Comment:
I'm not sure we should return here. If we do we're losing durability by not persisting the silences. How do you feel about logging and carrying on? Worst-case scenario we would just try again on the next interval and the problem might be fixed by then.

Reply:
The only underlying way Prom fails here is if there's a zero expiration timestamp. I kind of hate that, because once you get invalid data you're doomed to fail forever.

I agree that continuing and snapshotting after this error is what I should be doing.